src/base/rss/rss_autodownloader.cpp

```diff
@@ -52,6 +52,36 @@
 #include "rss_folder.h"
 #include "rss_session.h"
 
+namespace {
+    // find all occurrences of <placeholder> in original string, and replace them with value of "placeholder" from article
+    // example: savePathTpl = "c:\downloads\<attr_name> - <description>\something\<title>\<not_found_attr>"
+    //          article = <item>
+    //                      <title>some title</title>
+    //                      <link>some link</link>
+    //                      <guid isPermaLink="false">some-guid</guid>
+    //                      <pubDate>Thu, 02 Jun 2016 03:50:04 +0000</pubDate>
+    //                      <description>some description</description>
+    //                      <attr_name>some attr value</attr_name>
+    //                    </item>
+    //          result = "c:\downloads\some attr value - some description\something\some title\not_found_attr"
+    QString generateSavePath(const QString savePath, const QVariantHash articleData)
+    {
+        QRegExp rx("<(.+)>");
```

Comment:
I just picked the simplest syntax that would get my feature working for my use cases. Of course it can be revised and expanded, to include more complicated ones.
I am not an expert in regexp, nor am I a c++ programmer. Adding to that - I can't seem to compile the project currently on my computer, using MSVC2017 (is there anybody that can help me with that?). But if you have any specific ideas, I can give it a shot.